src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3659,6 +3660,42 @@ int journal_file_open(
         return r;
 }
 
+int journal_file_parse_uid(const char *path, uid_t *ret_uid, bool parse_archive) {
+        _cleanup_free_ char *buf = NULL;
+        _cleanup_free_ char *p = NULL;
+        const char *a, *b, *at;
+        int r;
+
+        assert(path);
+        assert(ret_uid);
+
+        r = path_extract_filename(path, &p);
+        if (r < 0)
+                return r;
+        if (r == O_DIRECTORY)
+                return -EISDIR;
+
+        a = startswith(p, "user-");
+        if (!a)
+                return -EREMOTE;
```

Comment:
EREMOTE seems a bit weird here, what about EINVAL?

Reply:
the question is: what is EREMOTE supposed to indicate? that this is not a user journal file, but a system one? If so, please add a comment saying so explicitly.

Reply:
Indeed, EREMOTE was supposed to indicate that this is not a journal file and can be anything else.

It also returns EREMOTE when archived journal files are excluded (with `parse_archived=true`) and the filename contains '@'. I guess we should make a distinction between the 2 cases so I propose to return EIO for the second case.